lib/sprockets/npm.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+require 'json'
+
+module Sprockets
+  module Npm
+    # Internal: Override resolve_alternates to install package.json behavior.
+    #
+    # load_path    - String environment path
+    # logical_path - String path relative to base
+    #
+    # Returns candiate filenames.
+    def resolve_alternates(load_path, logical_path)
+      candidates, deps = super
+
+      # package.json can only be nested one level deep
```

Comment:
Honestly I'm not sure, but that's my interpretation of `LOAD_AS_DIRECTORY(X)` that package.json is immediately inside of folder X not in subfolders:  

```
LOAD_AS_DIRECTORY(X)
1. If X/package.json is a file,
   a. Parse X/package.json, and look for "main" field.
   b. let M = X + (json main field)
   c. LOAD_AS_FILE(M)
```

How can we be 100% sure of this?